### PR TITLE
feat: yabloc test

### PR DESCRIPTION
**Pull request by hayato-m126**
## Types of PR

- [x] New Features

## Description

- add unit test

To add test code for quality control of driving_log_replayer

The driving_log_replayer is a tool that operates as follows:

1. It subscribes to topics published by Autoware.
2. It determines success or failure based on the subscribed topic meeting certain conditions and writes the results to result.jsonl.

Therefore, add unit tests for the process of writing messages received from Autoware.

1. Prepare a message for success and a message for failure.
2. Test whether the Result class can correctly determine success and failure by passing messages to it.

## How to review this PR

Thanks to colcon test, unit tests are now automatically executed. Therefore, please verify the following:

1. Ensure that all checks in GitHub Actions have passed.
2. Verify that the contents of test_yabloc.py are correctly coded.

### CODE REVIEW

File: driving_log_replayer/driving_log_replayer/yabloc.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        include_target_status = False
+        for diag_status in msg.status:
+            if diag_status.name != AvailabilityResult.TARGET_DIAG_NAME:
+                continue
+            include_target_status = True
+            values = {value.key: value.value for value in diag_status.values}
+            status_str = values.get("Availability", "Availability Key Not Found")  # avoid KeyError
+            self.success = status_str == "OK"
+            self.summary = f"{self.name} ({self.success_str()}): {status_str}"
+            break
```
Comment (hayato-m126) [inline driving_log_replayer/driving_log_replayer/yabloc.py:39]: @kminoda 
Is it possible for “yabloc_monitor: yabloc_status” to be included multiple times in DiagnosticArray? I have inserted a break with the intention of terminating the search once one is found.

File: driving_log_replayer/driving_log_replayer/yabloc.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        if include_target_status:
+            return {
+                "Ego": {},
+                "Availability": {
+                    "Result": self.success_str(),
+                    "Info": [],
+                },
+            }
+        return {
```
Comment (hayato-m126) [inline driving_log_replayer/driving_log_replayer/yabloc.py:48]: If “yabloc_monitor: yabloc_status” is not found in DiagnosticArray, the status will not be updated. The log will record that it was not found.

**Review by hayato-m126 — COMMENTED** [review]
add review comment

File: driving_log_replayer/driving_log_replayer/yabloc.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
```
Comment (hayato-m126) [inline driving_log_replayer/driving_log_replayer/yabloc.py:48]: If “yabloc_monitor: yabloc_status” is not found in DiagnosticArray, the status will not be updated. The log will record that it was not found.

File: driving_log_replayer/driving_log_replayer/yabloc.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+            break
+        if include_target_status:
+            return {
+                "Ego": {},
+                "Availability": {
+                    "Result": {"Total": self.success_str(), "Frame": self.success_str()},
+                    "Info": {},
+                },
+            }
+        return {
```
Comment (hayato-m126) [inline driving_log_replayer/driving_log_replayer/yabloc.py:48]: If “yabloc_monitor: yabloc_status” is not found in DiagnosticArray, the status will not be updated. The log will record that it was not found.

**Review by kminoda — APPROVED** [review]
LGTM

**Pull request merged**